servant-auth-server/src/Servant/Auth/Server/Internal/ConfigTypes.hs

```diff
@@ -79,7 +75,31 @@ defaultCookieSettings = CookieSettings
     , cookiePath        = Just "/"
     , cookieSameSite    = SameSiteLax
     , sessionCookieName = "JWT-Cookie"
-    , xsrfCookieName    = "XSRF-TOKEN"
+    , xsrfSetting       = XsrfCookie def
+    }
+
+data XsrfSetting = XsrfCookie XsrfCookieSettings
+                 | XsrfCheckOrigin (Maybe BS.ByteString -> Bool)
+                 deriving (Generic)
+
+
+-- | The policies to use when generating and verifying XSRF cookies
+data XsrfCookieSettings = XsrfCookieSettings
+  {
+  -- | What name to use for the cookie used for CSRF protection.
+    xsrfCookieName :: BS.ByteString
+  -- | What path to use for the cookie used for CSRF protection. Default @Just "/"@.
+  , xsrfCookiePath :: Maybe BS.ByteString
+  -- | What name to use for the header used for CSRF protection.
+  , xsrfHeaderName :: BS.ByteString
+  } deriving (Eq, Show, Generic)
```

Comment:
Should this have all the same settings as the session cookie? I think the suggestion in the README to have JS pull out the XSRF token from the cookies would not work if you use HTTP-only cookies. However, you really want to use HTTP-only cookies for the session cookie. Therefore having them the same puts you in a bind.